Subject: [PATCH] CONFIG_SWAP=n: include/linux/swap.h needs <linux/pagemap.h>

When compiling a m68k kernel with CONFIG_SWAP=n, I get:

    include/asm-generic/tlb.h: In function 'tlb_flush_mmu':
    include/asm-generic/tlb.h:76: error: implicit declaration of function 'release_pages'
    include/asm-generic/tlb.h: In function 'tlb_remove_page':
    include/asm-generic/tlb.h:105: error: implicit declaration of function 'page_cache_release'

because include/linux/swap.h defines free_page_and_swap_cache() and
free_pages_and_swap_cache() to page_cache_release() resp. release_pages()
without including <linux/pagemap.h>.

As the comment in include/linux/swap.h says it cannot include <linux/pagemap.h>
on sparc, 

---
 include/asm-generic/tlb.h |    5 +++++
 include/linux/swap.h      |    2 ++
 2 files changed, 7 insertions(+)

--- a/include/asm-generic/tlb.h
+++ b/include/asm-generic/tlb.h
@@ -14,6 +14,11 @@
 #define _ASM_GENERIC__TLB_H
 
 #include <linux/swap.h>
+
+#ifndef CONFIG_SWAP
+#include <linux/pagemap.h>
+#endif
+
 #include <asm/pgalloc.h>
 #include <asm/tlbflush.h>
 
--- a/include/linux/swap.h
+++ b/include/linux/swap.h
@@ -276,6 +276,8 @@ static inline void disable_swap_token(vo
 
 #else /* CONFIG_SWAP */
 
+//#include <linux/pagemap.h>
+
 #define total_swap_pages			0
 #define total_swapcache_pages			0UL
 
